is-wds-admin.php

```diff
@@ -110,104 +133,77 @@ public function get_cap_name() {
-				return true;
-			}
+		// Check if the current user is 'wds_admin'.
+		if ( ! is_wp_error( $current_user ) ) {
+			return $this->privileged_user === $current_user->user_login;
 		}
 
-		// Also allow for filtering globally,
-		return apply_filters( 'is_wds_admin', false, $current_user );
+		return false;
 	}
 
 	/**
-	 * Add the 'is_wds_admin' capability if it doesn't exist. Remove if the user should not have it.
+	 * Are we working with a WDS SSO User?
 	 *
-	 * @since 1.0.0
+	 * @author Aubrey Portwood
+	 * @since  1.1.0
+	 *
+	 * @link https://github.com/WebDevStudios/wds-sso The WDS SSO Project.
+	 *
+	 * @return boolean True if we are, false if not.
 	 */
-	public function add_cap_if_not_exists() {
+	private function is_wds_sso_user() {
+		if ( ! class_exists( '\WDS\SSO\User' ) ) {
 
-		// Check to see if the user is privileged
-		$is_priviledged = $this->is_privileged_user();
+			// The user class doesn't even exist.
+			return false;
+		}
 
-		// Check to see if the user currently has the is_wds_admin cap.
-		$has_cap = current_user_can( $this->get_cap_name() );
+		if ( ! property_exists( \WDS\SSO\app(), 'user' ) || ! method_exists( \WDS\SSO\app()->user, 'is_sso_user' ) ) {
 
-		// Check to see if the current user is defined as the privileged user and
-		// if they don't already have the 'is_wds_admin' capability.
-		if ( $is_priviledged && ! $has_cap ) {
-			// Add the cap to the user
-			return $this->modify_cap( 'add' );
+			// The class isn't configured like we thought.
+			return false;
 		}
 
-		// If the user is not privileged, but has the capability, let's remove it from them.
-		if ( ! $is_priviledged && $has_cap ) {
-			// Remove cap from user.
-			return $this->modify_cap( 'remove' );
-		}
+		return \WDS\SSO\app()->user->is_sso_user();
 	}
 
 	/**
-	 * Helper method to add or remove 'is_wds_admin' capabity for current user.
-	 * @param  string $action 'add' or 'remove'
+	 * Add the 'is_wds_admin' capability if it doesn't exist.
+	 *
+	 * @since 1.0.0
 	 */
-	public function modify_cap( $action ) {
+	public function add_cap_if_not_exists() {
 
-		// Get the WP_User object.
-		$user = new WP_User( get_current_user_id() );
+		// The current user.
+		$user = wp_get_current_user(); // Get the WP_User object.
 
-		// If we didn't get a user, bail out.
-		if ( empty( $user ) || ! is_object( $user ) ) {
-			return;
+		// Check to see if the current user is defined as the privileged user and if they don't already have the 'is_wds_admin' capability.
+		if ( $this->is_privileged_user() && ! $this->is_wds_admin() && is_a( $user, 'WP_User' ) ) {
+			$user->add_cap( 'is_wds_admin' ); // Add the cap.
+			return; // We did it.
 		}
 
-		// Depening on the action that is being passed in, add or remove the capability for the user.
-		switch ( $action ) {
-			case 'add':
-				$user->add_cap( $this->get_cap_name() );
-				break;
-			case 'remove':
-				$user->remove_cap( $this->get_cap_name() );
-				break;
+		if ( is_a( $user, 'WP_User' ) ) {
```

Comment:
Also, am I reading this right? Above on line 189,  you're adding the cap, but here, if user IS also a WP_User object, you're removing the cap? Pretty sure this is backwards and should go above line 189, unless I'm reading this wrong.